types/nodemailer-express-handlebars/index.d.ts

```diff
@@ -0,0 +1,40 @@
+// Type definitions for nodemailer-express-handlebars 4.0
+// Project: https://github.com/yads/nodemailer-express-handlebars
+// Definitions by: Gabriel de Oliveira <https://github.com/th3rius>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { create } from "express-handlebars";
+import { SentMessageInfo, Transporter } from "nodemailer";
+import * as Mail from "nodemailer/lib/mailer";
+
+declare function hbs(options: hbs.NodemailerExpressHandlebarsOptions): Mail.PluginFunction;
+
+declare namespace hbs {
+    type Exphbs = ReturnType<typeof create>;
+
+    type ExphbsOptions = Parameters<typeof create>[0];
+
+    interface TemplateOptions {
+        template?: string;
+        context?: any;
+    }
+
+    type HbsTransporter = Transporter & {
+        sendMail(
+            mailOptions: Mail.Options & TemplateOptions,
+            callback: (err: Error | null, info: SentMessageInfo) => void,
+        ): void;
+        sendMail(mailOptions: Mail.Options & TemplateOptions): Promise<SentMessageInfo>;
+    };
+
+    interface NodemailerExpressHandlebarsOptions {
+        viewEngine: Exphbs | ExphbsOptions;
+        viewPath: string;
+        extName?: string;
+    }
+
+    export { TemplateOptions, HbsTransporter, NodemailerExpressHandlebarsOptions };
+}
+
+export = hbs;
+export as namespace hbs;
```

Comment:
Is this required?  It looks like the underlying library only has the `export =` export.

Reply:
Nope, but the the `Transport` type in nodemailer isn't compatiple with this library, since the plugin expects additional properties, so I added a replacement type inside that namespace. Should I just remove it?

Reply:
I believe `hbs.HbsTransporter` will still work correctly in consuming code (e.g. the tests) without `export as namespace`.